CHANGELOG.md

```diff
@@ -14,6 +14,9 @@ changes.
   - Conway formatted transactions can be submitted to the `hydra-node`, while past eras are still supported (except deprecated features like protocol updates).
   - This includes support for `PlutusV3` scripts, but most of the governance-related features have no meaning in the Hydra L2.
 
+- **BREAKING** Changes to the `hydra-node` API `Greetings` type [#1622](https://github.com/cardano-scaling/hydra/pull/1622):
+  - Added head id information into the Greetings message.
```

Comment:
Why does this need to be a breaking change?

I would strongly suggest that we just add a `headId` field to the `Greetings` message when a head is open / we have the id. Exclude it or `null` if there is no head open / we do not know the id.

Reply:
The motivating use case here was for Hydra doom: the control plane would connect to the offline heads on startup; we wanted to know the head ID of the node we connected to so that we could include it in log messages etc, but there was no way for us to get this without streaming the entire history of the node (which may be several gigabytes), or waiting for some *other* message (which would not be guaranteed in any particular time window) in order to identify the head ID.

In our case, we were using offline heads, which *always* have a head ID from the start, so I actually didn't think about the case of when a head wasn't open yet; but if we connected to a head that wasn't open yet, then just following the transactions from that point onward would tell us once the head opened, so an optional field is probably sufficient.